### PR TITLE
docs(behavior_path_planner): update pull over docs

### PR DESCRIPTION
**Pull request by kosuke55**
## Description

updated pull_over docs related to
- https://github.com/autowarefoundation/autoware.universe/pull/873
- https://github.com/autowarefoundation/autoware.universe/pull/1534

please see 
- https://github.com/tier4/autoware.universe/blob/docs/update_pull_over_docs/planning/behavior_path_planner/README.md
- https://autowarefoundation.github.io/autoware.universe/pr-1578/planning/behavior_path_planner/

## Related links

- https://github.com/autowarefoundation/autoware.universe/pull/873
- https://github.com/autowarefoundation/autoware.universe/pull/1534

## Tests performed

not tested because just modifying docs.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].
- [ ] The PR has been properly tested.
- [ ] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: planning/behavior_path_planner/README.md [hunk 1]
```diff
@@ -195,54 +196,153 @@ The Pull Over module is activated when goal is in the shoulder lane. Ego-vehicle
 
 - Pull over request condition
 
-  - The goal is in shoulder lane
-  - The distance from ego-vehicle to the destination is long enough for pull over
-    - The distance required for a pull over is determined by the sum of the straight distance before pull over, the straight distance after pull over, and the minimum distance required for pull over
+  - The goal is in shoulder lane and the ego-vehicle is in road lane.
+  - The distance between the goal and ego-vehicle is somewhat close.
+    - It is shorter than `requet_length`(default: < `100m`).
```
Comment (tkimura4) [inline planning/behavior_path_planner/README.md:201]:
```suggestion
    - It is shorter than `request_length`(default: < `100m`).
```

File: planning/behavior_path_planner/README.md [hunk 2]
```diff
@@ -195,54 +196,153 @@ The Pull Over module is activated when goal is in the shoulder lane. Ego-vehicle
...
+
+[Video of how arc_forward_parking works](https://user-images.githubusercontent.com/39142679/178034280-4b6754fe-3981-4aee-b5e0-970f34563c6d.mp4)
+
+###### Parameters arc backward parking
+
+| Name                                     | Unit  | Type   | Description                                      | Default value |
+| :--------------------------------------- | :---- | :----- | :----------------------------------------------- | :------------ |
+| enable_arc_backward_parking              | [-]   | bool   | flag whether to enable arc backward parking      | false         |
+| after_backward_parking_straight_distance | [m]   | double | straight line distance after pull over end point | 2.0           |
+| backward_parking_velocity                | [m/s] | double | velocity when forward parking                    | -0.3          |
```
Comment (tkimura4) [inline planning/behavior_path_planner/README.md:337]:
```suggestion
| backward_parking_velocity                | [m/s] | double | velocity when backward parking                   | -0.3          |
```

File: planning/behavior_path_planner/README.md [hunk 3]
```diff
@@ -195,54 +196,153 @@ The Pull Over module is activated when goal is in the shoulder lane. Ego-vehicle
...
+| pull_over_velocity         | [m/s]  | double | decelerate to this speed by the goal search area                                                                                        | 2.0           |
+| pull_over_minimum_velocity | [m/s]  | double | speed of pull_over after stopping once. this prevents excessive acceleration.                                                           | 0.3           |
+| margin_from_boundary       | [m]    | double | distance margin from edge of the shoulder lane                                                                                          | 0.5           |
+| decide_path_distance       | [m]    | double | decide path if it approaches this distance relative to the parking position. after that, no path planning and goal search are performed | 10.0          |
+| maximum_deceleration       | [m/s2] | double | maximum deceleration. Maximum deceleration. it prevents sudden deceleration when a parking path cannot be found suddenly                | 0.5           |
```
Comment (tkimura4) [inline planning/behavior_path_planner/README.md:231]: The description of maximum_deceleration seems to be strange.
Maybe, `Maximum deceleration` is inserted incorrectly

Comment (kosuke55) [reply planning/behavior_path_planner/README.md]: thanks, fixed in https://github.com/autowarefoundation/autoware.universe/pull/1578/commits/0e7de761312528ac65945ae0e8292ed67ff146a3

File: planning/behavior_path_planner/README.md (hunk 2 above)
Comment (kosuke55) [reply planning/behavior_path_planner/README.md]: fixed in https://github.com/autowarefoundation/autoware.universe/pull/1578/commits/0e7de761312528ac65945ae0e8292ed67ff146a3

File: planning/behavior_path_planner/README.md (hunk 1 above)
Comment (kosuke55) [reply planning/behavior_path_planner/README.md]: fixed in https://github.com/autowarefoundation/autoware.universe/pull/1578/commits/0e7de761312528ac65945ae0e8292ed67ff146a3

File: planning/behavior_path_planner/README.md [hunk 4]
```diff
@@ -15,7 +15,7 @@ The following modules are currently supported:
 - **Lane Following**: Generate lane centerline from map.
 - **Lane Change**: Performs a lane change. This module is performed when it is necessary and a collision check with other vehicles is cleared.
 - **Obstacle Avoidance**: Perform an obstacle avoidance. This module is for avoidance of a vehicle parked on the edge of the lane or overtaking a low-speed obstacle.
-- **Pull Over**: Performs a pull over. This module is performed when goal is in the shoulder lane. Ego-vehicle will stop at the goal.
+- **Pull Over**: Performs a pull over. This module is performed when ego-vehicle is in the road lane and goal is in the shoulder lane. ego-vehicle will stop at the goal.
```
Comment (TakaHoribe) [inline planning/behavior_path_planner/README.md:18]:
```suggestion
- **Pull Over**: Performs a pull over. This module is performed when ego-vehicle is in the road lane and the goal is in the shoulder lane. The ego-vehicle will stop at the goal.
```

**Review by TakaHoribe — APPROVED** [review]
LGTM

**Pull request merged**